### DISCUSSION TITLE
Unable to reset password for devpi

### DISCUSSION THREAD

**Discussion by kgaainer** [devpi/devpi#1040] [Q&A]
Unable to reset root volume when using in a container. 

   devpi-server 6.9.2
    devpi-web 4.2.1
   devpi-client 7.0.2

cli:  devpi login root
password for user root at http://nodeip:nodeport/

401 Unauthorized: user 'root' could not be authenticated

Is there a way to reset the password?

**Comment by fschulze** [comment]
I don't understand the problem you have. Could you try to give some examples or error messages you get?

**Comment by kgaainer** [comment]
Sure, I am new to devpi, so this might sound basic. The devpi client runs on a container, however, I am unable to add a new user or make any changes since root password is unknown and the only user present is root.

#devpi index -l
no current authenticated user


#devpi login root --> This does prompt for a password but password is unknown. Is there a way to reset the root password ? 
401 Unauthorized: user 'root' could not be authenticated

Also the only admin user present is root.

#devpi user -l
root

**Comment by fschulze** [comment]
By default the ``root`` user has an empty password (hit return/enter during the prompt), see https://devpi.net/docs/devpi/devpi/stable/+doc/quickstart-server.html#setting-the-root-password

During initialization with the ``devpi-init`` command one can use ``--root-passwd`` or ``--root-passwd-hash`` to set an initial password.

There is also the ``devpi-passwd`` to change the password of a user.

I hope this helps.
